### PR TITLE
SNMP documentation to include ACL support

### LINKED ISSUE #377: ACL support for SNMP
**Issue Type**

- Template enhancement

**Is your feature request related to a problem? Please describe.**
We would require SNMP in AVD to support adding ACLs for security.

**Describe the solution you'd like**

`snmp-server ipv4 access-list ACL-NAME`

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by xaviramon**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [x] Documentation content changes
- [ ] Other (please describe):

## Related Issue(s)

- Fixes documentation missing on #377 
- Fixes #377 

## Checklist:

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://www.avd.sh/docs/contributing/) document.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [x] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).

### CODE REVIEW

**Comment by xaviramon** [comment]
Please, review. Since ACLs on SNMP were not considered as per VRF, this is a breaking change.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md [hunk 1]
```diff
@@ -424,8 +424,16 @@ Redundancy:
 snmp_server:
   contact: < contact_name >
   location: < location >
-  ipv4_access_list: < ipv4-access-list >
-  ipv6_access_list: < ipv6-access-list >
+  ipv4_ACLs:
```
Comment (onurgashi) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: Maybe use lowercase everywhere, as that's how we do everything

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md [hunk 2]
```diff
@@ -424,8 +424,16 @@ Redundancy:
 snmp_server:
   contact: < contact_name >
   location: < location >
-  ipv4_access_list: < ipv4-access-list >
-  ipv6_access_list: < ipv6-access-list >
+  ipv4_ACLs: 
+    - name: < ipv4-access-list >
+      vrf: < vrf >
```
Comment (onurgashi) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md:430]: Does this cover the default (global table) which doesn't have any vrf setting?
Can we reuse the same ACL in global and i.e MGMT?

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/snmp-settings.j2 [hunk 3]
```diff
@@ -1,9 +1,24 @@
 {% if snmp_server is defined and snmp_server is not none %}
 ### SNMP Configuration Summary
 
-| Contact | Location | SNMP Traps | IPv4 ACL | IPv6 ACL |
-| ------- | -------- | ---------- | -------- | -------- |
-| {{ snmp_server.contact | default('-') }} | {{ snmp_server.location | default('-') }} | {% if snmp_server.traps.enable is defined and snmp_server.traps.enable is not none %}{% if snmp_server.traps.enable %} Enabled {% else %} Disabled {% endif %}{% endif %} | {{ snmp_server.access_list_v4 | default('-') }} | {{ snmp_server.access_list_v6 | default('-') }} |
+| Contact | Location | SNMP Traps |
+| ------- | -------- | ---------- |
+| {{ snmp_server.contact | default('-') }} | {{ snmp_server.location | default('-') }} | {% if snmp_server.traps.enable is defined and snmp_server.traps.enable is not none %}{% if snmp_server.traps.enable %} Enabled {% else %} Disabled {% endif %}{% else %} Disabled {% endif %} |
```
Comment (onurgashi) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/snmp-settings.j2:6]: Double else?

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md (hunk 1 above)
Comment (xaviramon) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md]: Done. Thanks for the catch up!

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md (hunk 2 above)
Comment (xaviramon) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md:430]: Yes, no VRF implies default VRF, as the command in EOS. Also "default" can be used as VRF value.
Yes, same ACL can be reused in different VRFs but it needs to be defined twice:
  ipv4_ACLs: 
    - name: MyACL
    - name: MyACL
       vrf: MGMT

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/snmp-settings.j2 (hunk 3 above)
Comment (xaviramon) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/snmp-settings.j2:6]: Yes. Probably there is a better way to optimise but:
if snmp_server.traps.enable exists and defined:
   if snmp_server.traps.enable equals TRUE:
      print Enabled
   else //Variable is defined and FALSE
      print Disabled
   endif
else //From the first if. Basically, if the variable is not defined or empty
print Disabled
endif

Agree?

**Comment by onurgashi** [comment]
LGTM!

**Comment by titom73** [comment]
Doing rebase before merging

**Review by titom73 — APPROVED** [review]
LGTM

**Pull request merged**